contrib/Bright/SitePackage.lua

```diff
@@ -0,0 +1,90 @@
+--  a copy of this software and associated documentation files (the
+--  "Software"), to deal in the Software without restriction, including
+--  without limitation the rights to use, copy, modify, merge, publish,
+--  distribute, sublicense, and/or sell copies of the Software, and to
+--  permit persons to whom the Software is furnished to do so, subject
+--  to the following conditions:
+--
+--  The above copyright notice and this permission notice shall be
+--  included in all copies or substantial portions of the Software.
+--
+--  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+--  EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+--  OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+--  NONINFRINGEMENT.  IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+--  BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+--  ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+--  CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+--  THE SOFTWARE.
+--
+--------------------------------------------------------------------------
+
+require("strict")
+local hook   = require("Hook")
+local uname  = require("posix").uname
+
+-- By using the hook.register function, this function "load_hook" is called
+-- ever time a module is loaded with the file name and the module name.
+
+local s_msgA = {}
+
+function load_hook(t)
+   -- the arg t is a table:
+   --     t.modFullName:  the module full name: (i.e: gcc/4.7.2)
+   --     t.fn:           The file name: (i.e /apps/modulefiles/Core/gcc/4.7.2.lua)
+
+   -- Your site can use this any way that suits.  Here are some possibilities:
+   --  a) Write this information into a file in your users directory (say ~/.lmod.d/.save).
+   --     Then once a day/week/month collect this data.
+   --  b) have this function call syslogd to register that this module was loaded by this
+   --     user
+   --  c) Write the same information directly to some database.
+
+   -- This is an example writing to syslogd:
+
+   if (mode() ~= "load") then return end
+   local user        = os.getenv("USER")
+   local host        = uname("%n")
+   local currentTime = epoch()
+-- 20170503FG improve syslog 4 SGE jobs, but make it flexible, if variable is not there
+   local jobid       = os.getenv("JOB_ID")
+   if jobid == nil then
+      jobid = "unknown"
+   end
+   local msg         = string.format("USER=%s JOB_ID=%s MODULE=%s FULLPATH=%s HOST=%s TIME=%f",
+                                     user, jobid, t.modFullName, t.fn, host, currentTime)
+   local a           = s_msgA
+   a[#a+1]           = msg
+end
+
+hook.register("load",load_hook)
+
+
+function report_loads()
+
+   local sys         = os.getenv("LMOD_sys") or "Linux"
+   if (sys == "Linux") then
+      local a = s_msgA
+      for i = 1,#a do
+         local msg = a[i]
+         os.execute("logger -t ModuleUsageTracking -p local0.info " .. msg)
```

Comment:
you want add ` -- ` before appending message or you might get trouble with unloads (`-GCC`)